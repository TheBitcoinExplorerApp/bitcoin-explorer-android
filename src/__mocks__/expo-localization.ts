const getLocales = () => [
  { country: "pt", languageTag: "pt-BR", languageCode: "pt", isRTL: false },
];

export default getLocales;
