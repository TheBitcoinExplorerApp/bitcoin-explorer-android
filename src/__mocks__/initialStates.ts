import { BlockType } from "src/components/BlocksInfo/components/Block/type";
import { AddressInfoType, TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks: BlockType[] = [
  {
    blockHash: "",
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,

    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};

export const initialStateFormattedTransactions = [
  {
    transactionId: "0",
    value: 0,
    fee: 0,
  },
  {
    transactionId: "1",
    value: 0,
    fee: 0,
  },
];

export const initialStateAddress: AddressInfoType = {
  address: "",
  balance: 0,
  totalAmountReceived: 0,
  totalAmountSent: 0,
};
