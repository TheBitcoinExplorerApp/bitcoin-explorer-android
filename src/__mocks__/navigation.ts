const mockNavigation = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNavigation = jest.requireActual("@react-navigation/native");
  return {
    ...actualNavigation,
    useNavigation: () => ({
      navigate: mockNavigation,
    }),
  };
});
