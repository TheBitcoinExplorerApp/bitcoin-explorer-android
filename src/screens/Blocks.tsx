// eslint-disable-next-line import/no-extraneous-dependencies
import { ActivityIndicator } from "react-native";
import Main from "src/components/templates/Main";
import BlocksInfo from "src/components/BlocksInfo/BlocksInfo";
import useAppDataStore from "src/context/DataProvider";

type BlocksProps = {
  navigation: unknown;
};

export default function Blocks(props: BlocksProps) {
  const { navigation } = props;

  const { isLoading } = useAppDataStore();

  const showLoading = <ActivityIndicator size="large" color="white" />;

  return (
    <Main navigation={navigation} actualScreen="Blocks">
      {isLoading ? showLoading : <BlocksInfo />}
    </Main>
  );
}
