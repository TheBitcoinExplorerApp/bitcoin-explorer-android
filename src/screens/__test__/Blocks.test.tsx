// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import "src/__mocks__/navigation";
import Blocks from "../Blocks";

describe("Blocks", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Blocks navigation={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
