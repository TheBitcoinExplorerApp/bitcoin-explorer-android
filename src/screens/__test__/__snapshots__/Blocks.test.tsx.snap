// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks renders correctly 1`] = `
[
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
      }
    }
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    showsVerticalScrollIndicator={false}
    style={
      {
        "width": "100%",
      }
    }
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          {
            "backgroundColor": "#101429",
            "flex": 1,
            "gap": 18,
            "paddingHorizontal": 24,
            "paddingTop": 22.5,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 26,
              "justifyContent": "center",
            }
          }
        >
          <svg
            height={32}
          />
        </View>
        <View
          style={
            {
              "width": "100%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Pesquisar"
            placeholderTextColor="#8d8d9a"
            style={
              {
                "backgroundColor": "#1d2133",
                "borderRadius": 5,
                "color": "#FFF",
                "fontSize": 18,
                "padding": 12,
                "width": "100%",
              }
            }
            value=""
          />
        </View>
        <ActivityIndicator
          color="white"
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#111428",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "space-between",
            "paddingVertical": 5,
            "width": 90,
          }
        }
      >
        <svg
          style={
            {
              "color": "#D6D6D6",
            }
          }
        />
        <Text
          style={
            {
              "color": "#D6D6D6",
            }
          }
        >
          Home
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "space-between",
            "paddingVertical": 5,
            "width": 90,
          }
        }
      >
        <svg
          style={
            {
              "color": "#DF7800",
            }
          }
        />
        <Text
          style={
            {
              "color": "#DF7800",
            }
          }
        >
          Blocos
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "space-between",
            "paddingVertical": 5,
            "width": 90,
          }
        }
      >
        <svg
          style={
            {
              "color": "#D6D6D6",
            }
          }
        />
        <Text
          style={
            {
              "color": "#D6D6D6",
            }
          }
        >
          Transações
        </Text>
      </View>
    </View>
  </View>,
]
`;
